src/software/ai/intent/chip_intent.h

```diff
@@ -44,4 +44,14 @@ class ChipIntent : public ChipPrimitive, public Intent
      * @return true if the ChipIntents are not equal and false otherwise
      */
     bool operator!=(const ChipIntent& other) const;
+
+    /**
+     * Generates Primitive from this Intent
+     *
+     * @return The Primitive that represents this Intent
+     */
+    PrimitiveMsg generatePrimitive() const;
```

Comment:
Not sure how I feel about only some `Intents` hacing a `generatePrimitive` function. This is not consistent between all primitives, and to me makes it unclear what code is responsible for generating primitives. The Intents, or the Navigator.